### ISSUE TITLE
build URLs doesn't work with buildername

### ISSUE THREAD

**Issue by aakashja** [buildbot/buildbot#3465]
After fixing https://github.com/buildbot/buildbot/issues/3407, REST API works fine with builder names. Also some of the URLs work fine with builder name (e.g.: https://nine.buildbot.net/#/builders/buildbot-job would work).

However, some of the build URLs still doesn't work. For example, following URLs wouldn't work:
https://nine.buildbot.net/#/builders/buildbot-job/builds/1
https://nine.buildbot.net/#/builders/buildbot-job/builds

It seems like backend has the required support for these URLs, but the frontend is not able to handle buildername in URL properly. Some of the details are in https://github.com/buildbot/buildbot/pull/3410#issuecomment-316918060.

build URLs should work with buildername, so that people do not have to remember the builder id and can easily type the URL based on builder names.

**Comment by tardyp** [comment]
This needs to be resolved by making sure the performance is still good.

The problem is that update events are only sent to the ("builders", builderid, "builds") and not ("builders", buildername, "builds")

So supporting both builderid and buildername in the UI will double the number of events sent to the event system.

Also we need to make sure that the buildername is properly urlquoted in the URL.
We need to make sure that this also works with utf8 characters, as buildernames are not (despite wrongly documented in nine transition) identifiers.

**Comment by churlinpr** [comment]
Another issue is that using the builder's name doesn't work if it contains a space. The URL with spaces replaced by %20 yields an empty list (say https://nine.buildbot.net/#/builders/buildbot%20job if buildbot-job was named "buildbot job")

**Comment by tardyp** [comment]
Another option would be to use the
builderid API only after querying the buildername, if we detect that the buildid in URL is not a number.

This makes the page a little bit slower (one more serialised API query), but is a simple solution for this long requested feature.
Especially if we use the builder cache for that this could be non noticable.

**Comment by XonqNopp** [comment]
Any update on this? It would be so much better that we can substitute the builder ID with the builder name so you know where you will land when you open the URL...

cc @gzzi @99rgosse
